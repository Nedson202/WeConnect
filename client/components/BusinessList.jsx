import 'rc-pagination/assets/index.css';
import React, { Component } from 'react';
import Pagination from 'rc-pagination';
import BusinessCard from './BusinessCard';

/**
 * @class BusinessList
 * 
 * @extends {Component}
 */
class BusinessList extends Component {
  /**
   * @description Creates an instance of BusinessList.
   * 
   * @param {object} props 
   * 
   * @memberof Profile
   */
  constructor(props){
    super(props);

    this.state = {
      current: 1
    };

    this.onChange = this.onChange.bind(this);
  }

  /**
   * @description Toggler for pagination
   * 
   * @param {any} page
   * 
   * @returns {undefined}
   * 
   * @memberof BusinessList
   */
  onChange(page) {
    const genPage = `page=${page}`
    
    this.setState({
      current: page,
    });

    if(location.pathname == '/dashboard') {
      this.props.fetchBusinessesByUserId(genPage);      
    }

    if(location.pathname == '/businesses') {
      this.props.fetchBusinesses(genPage);
    }
  }

  /**
   * @description Renders the component to the dom
   * 
   * @returns {object} JSX object
   * 
   * @memberof BusinessList
   */
  render() {
    const { businesses, paginate } = this.props;

    let noBusiness;
    
    if(location.pathname !== '/searchresult') {
      noBusiness = (
        <h2 className="text-center" id="no-business-found">No business registered yet</h2>
      );
    }


    if(location.pathname === '/searchresult') {
      noBusiness = (
        <h2 className="text-center" id="no-business-found">No search request</h2>
      );
    }

    const businessList = (
      <div className="row">
        { businesses.map(business => <BusinessCard business={business} key={business.id} />)}
      </div>
    );

    const paginateAll = (
      <div>{ businesses.length > 0 ?
        <div className="d-flex justify-content-center">
          { paginate && paginate.limit === undefined ? null : 
          <Pagination 
            onChange={this.onChange} 
            current={this.state.current} 
            total={!paginate ? null : paginate.totalBusinesses} 
            defaultCurrent={this.state.current}
            defaultPageSize={!paginate ?  null : paginate.limit}
          />
          }
        </div> : null}
      </div>
    );

    return (
      <div>
        <div className="container card-container">
          {businesses.length === 0 ? noBusiness : businessList}
        </div>

        { !paginate ? null
          : paginate.totalBusinesses <= paginate.limit ? null :  paginateAll}
      </div>
    );
  }
}

export default BusinessList;
