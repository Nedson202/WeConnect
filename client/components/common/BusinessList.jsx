import 'rc-pagination/assets/index.css';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pagination from 'rc-pagination';
import BusinessCard from './BusinessCard';
import green from '../../images/default.jpeg';

/**
 * @class BusinessList
 *
 * @extends {Component}
 */
class BusinessList extends Component {
  /**
   * @description Creates an instance of BusinessList.
   *
   * @param {object} props
   *
   * @memberof Profile
   */
  constructor(props) {
    super(props);

    this.state = {
      current: 1
    };

    this.onChange = this.onChange.bind(this);
  }

  /**
   * @description Toggler for pagination
   *
   * @param {any} page
   *
   * @returns {undefined}
   *
   * @memberof BusinessList
   */
  onChange(page) {
    const genPage = `page=${page}`;
    const {
      queryData, filterBusiness, fetchBusinesses, fetchBusinessesByUserId
    } = this.props;
    const { pathname } = window.location;

    this.setState({
      current: page,
    });

    if (pathname === '/searchresult') {
      filterBusiness(queryData.option, queryData.query, genPage);
    }

    if (pathname === '/dashboard') {
      fetchBusinessesByUserId(genPage);
    }

    if (pathname === '/businesses' || window.location.pathname === '/adminpanel') {
      fetchBusinesses(genPage);
    }
  }
  /**
   * @description Deletes business from the database
   *
   * @param {any} event
   *
   * @returns {undefined}
   *
   * @memberof BusinessProfile
   */
  imageGenerator() {
    const { image } = this.state;
    if (!this.props.business && image) {
      return image;
    }
    if (image) {
      return image;
    }
    return green;
  }

  /**
   * @description Renders the component to the dom
   *
   * @returns {object} JSX object
   *
   * @memberof BusinessList
   */
  render() {
    const { businesses, paginate } = this.props;
    const { onChange, state } = this;
    const { pathname } = window.location;
    const { limit, totalBusinesses } = paginate;

    let noBusiness;

    if (pathname !== '/searchresult') {
      noBusiness = (
        <h2 className="text-center" id="no-business-found">No business registered yet</h2>
      );
    }

    if (pathname === '/searchresult') {
      noBusiness = (
        <h2 className="text-center" id="no-business-found">No Result Found</h2>
      );
    }

    const businessList = (
      <div className="row">
        { businesses.map(business => <BusinessCard business={business} key={business.id} />)}
      </div>
    );

    const paginateAll = (
      <div>{ businesses.length > 0 ?
        <div className="d-flex justify-content-center pagination">
          { paginate && totalBusinesses === undefined ? null :
          <Pagination
            onChange={onChange}
            current={state.current}
            total={!paginate ? null : totalBusinesses}
            defaultCurrent={state.current}
            defaultPageSize={!paginate ? null : limit}
          />
          }
        </div> : null}
      </div>
    );

    return (
      <div>
        <div className="container card-container">
          {businesses.length === 0 ? noBusiness : businessList}
        </div>

        <div>
          {
            !paginate ? null :
            <span>
              {totalBusinesses <= limit ? null : paginateAll}
            </span>
          }
        </div>
      </div>
    );
  }
}

BusinessList.propTypes = {
  filterBusiness: PropTypes.func,
  fetchBusinesses: PropTypes.func,
  fetchBusinessesByUserId: PropTypes.func,
  businesses: PropTypes.array.isRequired,
  business: PropTypes.object,
  paginate: PropTypes.object,
  queryData: PropTypes.object
};

BusinessList.defaultProps = {
  queryData: {},
  business: {},
  filterBusiness: null,
  fetchBusinessesByUserId: null,
  fetchBusinesses: null,
  paginate: {}
};

export default BusinessList;
