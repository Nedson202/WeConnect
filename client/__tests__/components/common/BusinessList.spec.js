import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import BusinessList from '../../../components/common/BusinessList';
import businessData from '../../__mockData__/business';

const { allBusiness, business, paginationResult } = businessData;
configure({ adapter: new Adapter() });

let props, pathname;
const setup = () => {
  window = {
    location: {
      // pathname: pathname
    }
  }
  props = {
    filterBusiness: jest.fn(() => Promise.resolve()),
    fetchBusinesses: jest.fn(() => Promise.resolve()),
    fetchBusinessesByUserId: jest.fn(() => Promise.resolve()),
    businesses: allBusiness,
    business: business,
    paginate: paginationResult,
    queryData: business,
    // pathname: '/searchresult'
  };

  return shallow(<BusinessList {...props} />);
};


describe('Component: BusinessList', () => {
  beforeEach(() => {
    global.window = {
      window: () => {},
    }
  })

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(6);
    expect(wrapper.find('span').length).toBe(1);
    // expect(wrapper.find('Pagination').length).toBe(1);

    wrapper.setProps({ businesses: [] });
    expect(wrapper.length).toBe(1);
    wrapper.setProps({ paginate: {}, business: [] });
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('div').length).toBe(4);
    expect(wrapper.find('Pagination').length).toBe(0);
    wrapper.setProps({ paginate: {
      totalBusiness: 3,
      limit: 5
    }, business: [] });
    expect(wrapper.find('div').length).toBe(4);
    wrapper.setProps({ paginate: {
      totalBusiness: undefined,
      limit: 5
    }, business: [] });
    expect(wrapper.find('Pagination').length).toBe(0);
    expect(toJSON(wrapper)).toMatchSnapshot();
  }) 
});