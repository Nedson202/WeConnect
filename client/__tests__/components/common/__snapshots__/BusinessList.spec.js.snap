// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BusinessList should render 1`] = `
<div>
  <div
    className="container card-container"
  >
    <h2
      className="text-center"
      id="no-business-found"
    >
      No business registered yet
    </h2>
  </div>
  <div>
    <span>
      <div />
    </span>
  </div>
</div>
`;
